app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -0,0 +1,161 @@
+package net.squanchy.schedule
+
+import android.content.Context
+import android.support.design.widget.CoordinatorLayout
+import android.support.design.widget.TabLayout
+import android.support.v4.view.ViewPager
+import android.support.v7.widget.Toolbar
+import android.text.Spanned
+import android.util.AttributeSet
+import android.view.View
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import net.squanchy.R
+import net.squanchy.analytics.Analytics
+import net.squanchy.analytics.ContentType
+import net.squanchy.home.Loadable
+import net.squanchy.navigation.Navigator
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.schedule.domain.view.Schedule
+import net.squanchy.schedule.view.ScheduleViewPagerAdapter
+import net.squanchy.support.font.applyTypeface
+import net.squanchy.support.font.getFontFor
+import net.squanchy.support.font.hasTypefaceSpan
+import net.squanchy.support.unwrapToActivityContext
+import timber.log.Timber
+
+
+class SchedulePageView @JvmOverloads constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int = 0) : CoordinatorLayout(
+        context,
+        attrs,
+        defStyleAttr
+), Loadable {
+
+    private val viewPagerAdapter: ScheduleViewPagerAdapter
+    private val service: ScheduleService
+    private val navigate: Navigator
+    private val analytics: Analytics
+    private lateinit var progressBar: View
+    private var subscription: Disposable? = null
```

Comment:
You can make it a non-null `CompositeDisposable` and init here

Reply:
6018f9008e78685bef7adab9884cf04ba6e49af0